editor/connections_dialog.cpp

```diff
@@ -365,6 +365,25 @@ void ConnectDialog::init(ConnectionData p_cd, bool p_edit) {
 
 	deferred->set_pressed(b_deferred);
 	oneshot->set_pressed(b_oneshot);
+
+	MethodInfo r_signal;
+	Ref<Script> source_script = source->get_script();
+	if (source_script.is_valid() && source_script->has_script_signal(signal)) {
+		List<MethodInfo> signals;
+		source_script->get_script_signal_list(&signals);
+		for (MethodInfo &mi : signals) {
+			if (mi.name == signal) {
+				r_signal = mi;
+				break;
+			}
+		}
+	} else {
+		ClassDB::get_signal(source->get_class(), signal, &r_signal);
+	}
+
+	int signal_argument_size = r_signal.arguments.size();
+	unbind_count->set_max(signal_argument_size);
```

Comment:
```suggestion
	unbind_count->set_max(r_signal.arguments.size());
```
Sorry for not noticing it earlier, but this variable is actually not necessary at all.